packages/template-retail-react-app/app/pages/checkout/partials/payment.jsx

```diff
@@ -4,67 +4,140 @@
 import {useCheckout} from '../util/checkout-context'
-import usePaymentForms from '../util/usePaymentForms'
-import {getCreditCardIcon} from '../../../utils/cc-utils'
+import {getPaymentInstrumentCardType, getCreditCardIcon} from '../../../utils/cc-utils'
+import {isMatchingAddress} from '../../../utils/utils'
 import {ToggleCard, ToggleCardEdit, ToggleCardSummary} from '../../../components/toggle-card'
-import PaymentSelection from './payment-selection'
+import PaymentForm from './payment-form'
 import ShippingAddressSelection from './shipping-address-selection'
 import AddressDisplay from '../../../components/address-display'
 import {PromoCode, usePromoCode} from '../../../components/promo-code'
+import {API_ERROR_MESSAGE} from '../../../constants'
 
 const Payment = () => {
     const {formatMessage} = useIntl()
+    const {data: basket} = useCurrentBasket()
+    const selectedShippingAddress = basket?.shipments && basket?.shipments[0]?.shippingAddress
+    const selectedBillingAddress = basket?.billingAddress
+    const appliedPayment = basket?.paymentInstruments && basket?.paymentInstruments[0]
+    const [billingSameAsShipping, setBillingSameAsShipping] = useState(
+        isMatchingAddress(selectedBillingAddress, selectedShippingAddress)
+    )
+    const {mutateAsync: addPaymentInstrumentToBasket} = useShopperBasketsMutation(
+        'addPaymentInstrumentToBasket'
+    )
+    const {mutateAsync: updateBillingAddressForBasket} = useShopperBasketsMutation(
+        'updateBillingAddressForBasket'
+    )
+    const {mutateAsync: removePaymentInstrumentFromBasket} = useShopperBasketsMutation(
+        'removePaymentInstrumentFromBasket'
+    )
+    const showToast = useToast()
+    const showError = () => {
+        showToast({
+            title: formatMessage(API_ERROR_MESSAGE),
+            status: 'error'
+        })
+    }
+
+    const {step, STEPS, goToStep, goToNextStep} = useCheckout()
 
-    const {
-        step,
-        checkoutSteps,
-        setCheckoutStep,
-        selectedShippingAddress,
-        selectedBillingAddress,
-        selectedPayment,
-        getPaymentMethods,
-        removePayment
-    } = useCheckout()
-
-    const {
-        paymentMethodForm,
-        billingAddressForm,
-        billingSameAsShipping,
-        setBillingSameAsShipping,
-        reviewOrder
-    } = usePaymentForms()
+    const billingAddressForm = useForm({
+        mode: 'onChange',
+        shouldUnregister: false,
+        defaultValues: {...selectedBillingAddress}
+    })
 
     const {removePromoCode, ...promoCodeProps} = usePromoCode()
 
-    useEffect(() => {
-        getPaymentMethods()
-    }, [])
+    const paymentMethodForm = useForm()
+
+    const onPaymentSubmit = async (formValue) => {
+        // The form gives us the expiration date as `MM/YY` - so we need to split it into
+        // month and year to submit them as individual fields.
+        const [expirationMonth, expirationYear] = formValue.expiry.split('/')
```

Comment:
Does the form validate the `MM/YY` format, or should we do that here?

Reply:
yes the form does!